Description: Fix makefile architecture
 Modify the top-level makefile, to select the correct
 architecture based on DEB_BUILD_ARCH
Author: Daniel Lintott <daniel@serverb.co.uk>
Forwarded: no
---
Index: dynamips/Makefile
===================================================================
--- dynamips.orig/Makefile	2014-02-10 20:03:09.000000000 +0000
+++ dynamips/Makefile	2014-02-10 20:22:36.852992794 +0000
@@ -4,12 +4,12 @@
 #   - Use "x86" for a build on x86 (32-bits)
 #   - Use "amd64" for a build on x86_64 (64-bits)
 #   - Use "nojit" for unsupported architectures.
-ifeq ($(shell arch),x86_64)
-export DYNAMIPS_ARCH?=amd64
-else
-ifeq ($(shell arch),i686)
+ifeq ($(shell dpkg-architecture -qDEB_HOST_ARCH_CPU),i386)
 export DYNAMIPS_ARCH?=x86
 else
+ifeq ($(shell dpkg-architecture -qDEB_HOST_ARCH_CPU),amd64)
+export DYNAMIPS_ARCH?=amd64
+else
 export DYNAMIPS_ARCH?=nojit
 endif
 endif
